### PR TITLE
r/storagegateway_gateway - smb file share visibility 

### LINKED ISSUE #18058: aws_storagegateway_gateway - Share Visibility Toggle does not appear to be available.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In the AWS storage gateway console.  Click "Actions" | "Edit SMB Settings".
At the bottom of the pop up window there is a check box for "Visibility Status".  If this check box is checked, the file share is then visible on the storage gateway.  As of the time of this writing, I don't see a means to manipulating this via the Terraform API.  It looks like it was just recently added to AWS PowerShell and AWS CLI.  Is there an ETA on adding this feature?

### New or Affected Resource(s)

aws_storagegateway_gateway | smb settings specifically.

### Potential Terraform Configuration

```hcl
resource "aws_storagegateway_gateway" "storage_gateway" {
  share_visible                 = true      **<<<<<<<<<<<<<<<<<<<<<<< Syntax similar to this for visibility toggle**
  gateway_ip_address            = var.ip_address
  gateway_name                  = var.machine_name
  gateway_timezone              = var.timezone
  gateway_type                  = "FILE_S3"
  cloudwatch_log_group_arn      = local.cw_log_group_arn
  smb_security_strategy         = var.smb_security_strategy
  smb_active_directory_settings {
    domain_name        = var.domain_name
    username           = var.domain_username
    password           = var.domain_password
    timeout_in_seconds = var.timeout_in_seconds
  }

  lifecycle {
    ignore_changes = [gateway_ip_address]
  }

  tags = merge(var.tags,
    {
      "Name" = var.solution_name
  })
}
```

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #18058

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
--- PASS: TestAccAWSStorageGatewayGateway_SMBVisibility (432.84s)
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (380.31s)
```

### CODE REVIEW

File: aws/resource_aws_storagegateway_gateway.go [hunk 1]
```diff
@@ -391,6 +395,19 @@ func resourceAwsStorageGatewayGatewayCreate(d *schema.ResourceData, meta interfa
...
+	if v, ok := d.GetOk("smb_file_share_visibility"); ok {
+		input := &storagegateway.UpdateSMBFileShareVisibilityInput{
+			GatewayARN:        aws.String(d.Id()),
+			FileSharesVisible: aws.Bool(v.(bool)),
+		}
+
+		log.Printf("[DEBUG] Storage Gateway Gateway %q setting SMB File Share Visibility", input)
+		_, err := conn.UpdateSMBFileShareVisibility(input)
+		if err != nil {
+			return fmt.Errorf("error setting SMB File Share Visibility: %w", err)
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_gateway.go:407]: I see that most of the current error messages for this resource don't include the ID but let's take this opportunity for improvement.

```suggestion
			return fmt.Errorf("error updating Storage Gateway Gateway (%s) SMB file share visibility: %w", d.Id(), err)
```

File: aws/resource_aws_storagegateway_gateway.go [hunk 2]
```diff
@@ -615,6 +633,19 @@ func resourceAwsStorageGatewayGatewayUpdate(d *schema.ResourceData, meta interfa
...
+	if d.HasChange("smb_file_share_visibility") {
+		input := &storagegateway.UpdateSMBFileShareVisibilityInput{
+			GatewayARN:        aws.String(d.Id()),
+			FileSharesVisible: aws.Bool(d.Get("smb_file_share_visibility").(bool)),
+		}
+
+		log.Printf("[DEBUG] Storage Gateway Gateway %q updating SMB File Share Visibility", input)
+		_, err := conn.UpdateSMBFileShareVisibility(input)
+		if err != nil {
+			return fmt.Errorf("error updating SMB File Share Visibility: %w", err)
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_gateway.go:645]:
```suggestion
			return fmt.Errorf("error updating Storage Gateway Gateway (%s) SMB file share visibility: %w", d.Id(), err)
```

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSStorageGatewayGateway_' ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 3 -run=TestAccAWSStorageGatewayGateway_ -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== RUN   TestAccAWSStorageGatewayGateway_tags
=== PAUSE TestAccAWSStorageGatewayGateway_tags
=== RUN   TestAccAWSStorageGatewayGateway_GatewayName
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayName
=== RUN   TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== PAUSE TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== RUN   TestAccAWSStorageGatewayGateway_GatewayTimezone
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayTimezone
=== RUN   TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== RUN   TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== PAUSE TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== RUN   TestAccAWSStorageGatewayGateway_SMBVisibility
=== PAUSE TestAccAWSStorageGatewayGateway_SMBVisibility
=== RUN   TestAccAWSStorageGatewayGateway_disappears
=== PAUSE TestAccAWSStorageGatewayGateway_disappears
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthUpload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthUpload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthDownload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthDownload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthAll
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthAll
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthAll
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (226.22s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthDownload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (288.17s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthUpload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (304.79s)
=== CONT  TestAccAWSStorageGatewayGateway_disappears
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (289.52s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBVisibility
--- PASS: TestAccAWSStorageGatewayGateway_disappears (182.13s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (812.62s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
--- PASS: TestAccAWSStorageGatewayGateway_SMBVisibility (276.48s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayName
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (213.26s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (219.42s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (263.33s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayTimezone
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (222.12s)
=== CONT  TestAccAWSStorageGatewayGateway_CloudWatchLogs
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (381.85s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Vtl
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (219.95s)
=== CONT  TestAccAWSStorageGatewayGateway_tags
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (250.34s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Stored
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (785.05s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_tags (269.16s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (198.44s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (209.35s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1920.922s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSStorageGatewayGateway_' ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 3 -run=TestAccAWSStorageGatewayGateway_ -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== RUN   TestAccAWSStorageGatewayGateway_tags
=== PAUSE TestAccAWSStorageGatewayGateway_tags
=== RUN   TestAccAWSStorageGatewayGateway_GatewayName
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayName
=== RUN   TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== PAUSE TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== RUN   TestAccAWSStorageGatewayGateway_GatewayTimezone
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayTimezone
=== RUN   TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== RUN   TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== PAUSE TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== RUN   TestAccAWSStorageGatewayGateway_SMBVisibility
=== PAUSE TestAccAWSStorageGatewayGateway_SMBVisibility
=== RUN   TestAccAWSStorageGatewayGateway_disappears
=== PAUSE TestAccAWSStorageGatewayGateway_disappears
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthUpload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthUpload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthDownload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthDownload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthAll
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthAll
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthAll
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (227.51s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthDownload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (251.66s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthUpload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (246.93s)
=== CONT  TestAccAWSStorageGatewayGateway_disappears
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (233.37s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBVisibility
--- PASS: TestAccAWSStorageGatewayGateway_disappears (216.29s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
--- PASS: TestAccAWSStorageGatewayGateway_SMBVisibility (271.14s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (200.92s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayName
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (922.97s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (210.40s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (232.71s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayTimezone
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (364.29s)
=== CONT  TestAccAWSStorageGatewayGateway_CloudWatchLogs
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (213.67s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Vtl
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (186.16s)
=== CONT  TestAccAWSStorageGatewayGateway_tags
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (199.92s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Stored
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (803.78s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (192.54s)
--- PASS: TestAccAWSStorageGatewayGateway_tags (237.75s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (229.47s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1959.315s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1). Note that Simple AD test failures are a known issue and are covered here: #16172

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSStorageGatewayGateway_' ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 3 -run=TestAccAWSStorageGatewayGateway_ -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== RUN   TestAccAWSStorageGatewayGateway_tags
=== PAUSE TestAccAWSStorageGatewayGateway_tags
=== RUN   TestAccAWSStorageGatewayGateway_GatewayName
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayName
=== RUN   TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== PAUSE TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== RUN   TestAccAWSStorageGatewayGateway_GatewayTimezone
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayTimezone
=== RUN   TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== RUN   TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== PAUSE TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== RUN   TestAccAWSStorageGatewayGateway_SMBVisibility
=== PAUSE TestAccAWSStorageGatewayGateway_SMBVisibility
=== RUN   TestAccAWSStorageGatewayGateway_disappears
=== PAUSE TestAccAWSStorageGatewayGateway_disappears
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthUpload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthUpload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthDownload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthDownload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthAll
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthAll
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== CONT  TestAccAWSStorageGatewayGateway_GatewayName
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
    resource_aws_storagegateway_gateway_test.go:400: Step 1/2 error: Error running apply: exit status 1
        2021/03/17 11:05:16 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.13.0

        Error: ClientException: Simple AD directory creation is currently not supported in this region. : RequestId: a7dcb8ae-b4b2-4c01-9f0e-d1cede25e138 : RequestId: a7dcb8ae-b4b2-4c01-9f0e-d1cede25e138
        {
          RespMetadata: {
            StatusCode: 400,
            RequestID: "a7dcb8ae-b4b2-4c01-9f0e-d1cede25e138"
          },
          Message_: "Simple AD directory creation is currently not supported in this region. : RequestId: a7dcb8ae-b4b2-4c01-9f0e-d1cede25e138 : RequestId: a7dcb8ae-b4b2-4c01-9f0e-d1cede25e138",
          RequestId: "a7dcb8ae-b4b2-4c01-9f0e-d1cede25e138"
        }

          on terraform_plugin_test.tf line 83, in resource "aws_directory_service_directory" "test":
          83: resource "aws_directory_service_directory" "test" {


--- FAIL: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (21.19s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (198.37s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayTimezone
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (227.81s)
=== CONT  TestAccAWSStorageGatewayGateway_CloudWatchLogs
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (260.54s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Vtl
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (202.36s)
=== CONT  TestAccAWSStorageGatewayGateway_tags
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (192.47s)
=== CONT  TestAccAWSStorageGatewayGateway_disappears
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (167.69s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthAll
--- PASS: TestAccAWSStorageGatewayGateway_disappears (169.39s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthDownload
--- PASS: TestAccAWSStorageGatewayGateway_tags (207.69s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthUpload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (247.40s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (220.61s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBVisibility
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (207.73s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Stored
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (215.97s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (180.40s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_SMBVisibility (229.70s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
    resource_aws_storagegateway_gateway_test.go:430: Step 1/2 error: Error running apply: exit status 1
        2021/03/17 11:22:34 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.13.0

        Error: ClientException: Simple AD directory creation is currently not supported in this region. : RequestId: 6ec6c8d6-1945-409a-8f28-7ba1ecf44ffa : RequestId: 6ec6c8d6-1945-409a-8f28-7ba1ecf44ffa
        {
          RespMetadata: {
            StatusCode: 400,
            RequestID: "6ec6c8d6-1945-409a-8f28-7ba1ecf44ffa"
          },
          Message_: "Simple AD directory creation is currently not supported in this region. : RequestId: 6ec6c8d6-1945-409a-8f28-7ba1ecf44ffa : RequestId: 6ec6c8d6-1945-409a-8f28-7ba1ecf44ffa",
          RequestId: "6ec6c8d6-1945-409a-8f28-7ba1ecf44ffa"
        }

          on terraform_plugin_test.tf line 83, in resource "aws_directory_service_directory" "test":
          83: resource "aws_directory_service_directory" "test" {


--- FAIL: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (18.97s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (234.61s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (179.87s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	1179.367s
FAIL
make: *** [testacc] Error 1
```

**Pull request merged**